src/viam/examples/modules/complex/gizmo/api.cpp

```diff
@@ -194,10 +196,6 @@ grpc::Status GizmoServer::DoTwo(::grpc::ServerContext* context,
     return grpc::Status();
 }
 
-void GizmoServer::register_server(std::shared_ptr<Server> server) {
-    server->register_service(this);
-}
```

Comment:
`register_server` is now useless, as `register_service` is called in the `ResourceServer` constructor.

Reply:
But the point of the virtual `register_server` was exactly to work around the issue you called out above, that up in `ResourceBase` the `this` pointer is not useable.

Reply:
I'd agree that `register_server()` used to be the way that you could register a service for a `Server`; my "replacement" for this logic is to pass `this` up to `ResourceServerBase` (which is also not ideal). I don't mean to say this was a bad design or anything, just that `register_service()` gets called in `ResourceServerBase` construction now. Did you have a reason to keep these methods around?